packages/smart-contracts/test/contracts/ProxyChainlinkConversionPath.js

```diff
@@ -0,0 +1,223 @@
+        const toDiffBalance = bigNumberify(toNewBalance.toString()).sub(toOldBalance.toString()).toString();
+        const feeDiffBalance = bigNumberify(feeNewBalance.toString()).sub(feeOldBalance.toString()).toString();
+      
+        // Check balance changes
+        expect(fromDiffBalance.toString()).to.equals('-' + bigNumberify(conversionToPay.result.toString()).add(conversionFees.result.toString()));
+        expect(toDiffBalance).to.equals(conversionToPay.result.toString());
+        expect(feeDiffBalance).to.equals(conversionFees.result.toString());
+      });
+      it('allows to transfer DAI tokens for EUR payment', async function () {
+        const path = [EUR_address, USD_address, DAI_address];
+        await testERC20.approve(testProxyChainlinkConversionPath.address, thousandWith18Decimal, { from });
+
+        const fromOldBalance = await testERC20.balanceOf(from);
+        const toOldBalance = await testERC20.balanceOf(to);
+        const feeOldBalance = await testERC20.balanceOf(feeAddress);
+        const conversionToPay = await chainlinkConversionPath.getConversion(
+          smallAmountInFIAT,
+          path
+        );
+        const conversionFees = await chainlinkConversionPath.getConversion(
+          smallerAmountInFIAT,
+          path
+        );
+    
+        const { logs } = await testProxyChainlinkConversionPath.transferFromWithReferenceAndFee(
+          to,
+          smallAmountInFIAT,
+          path,
+          referenceExample,
+          smallerAmountInFIAT,
+          feeAddress,
+          hundredWith18Decimal,
+          0,
+          { from },
+        );
+       
+        expectEvent.inLogs(logs, 'TransferWithConversionAndReference', {
+          amount: smallAmountInFIAT,
+          currency: path[0],
+          paymentReference: ethers.utils.keccak256(referenceExample),
+          feeAmount: smallerAmountInFIAT,
+          maxRateTimespan: "0",
+        });
+      
+      
+        const fromNewBalance = await testERC20.balanceOf(from);
+        const toNewBalance = await testERC20.balanceOf(to);
+        const feeNewBalance = await testERC20.balanceOf(feeAddress);
+      
+        const fromDiffBalance = bigNumberify(fromNewBalance.toString()).sub(fromOldBalance.toString()).toString();
+        const toDiffBalance = bigNumberify(toNewBalance.toString()).sub(toOldBalance.toString()).toString();
+        const feeDiffBalance = bigNumberify(feeNewBalance.toString()).sub(feeOldBalance.toString()).toString();
+      
+        // Check balance changes
+        expect(fromDiffBalance.toString()).to.equals('-' + bigNumberify(conversionToPay.result.toString()).add(conversionFees.result.toString()));
+        expect(toDiffBalance).to.equals(conversionToPay.result.toString());
+        expect(feeDiffBalance).to.equals(conversionFees.result.toString());
+      });
+    });
+
+    describe('transferFromWithReferenceAndFee with errors', () => {
+      it('cannot transfer with no approval', async function () {
```

Comment:
I think that's related to ERC20 interface, not to our contract? I don't understand why we would test it.

Reply:
Good point. Should we remove the test ?

Reply:
Yep, unless you see a point